runtime/gc_glue_java/ArrayletObjectModel.hpp

```diff
@@ -530,6 +540,20 @@ class GC_ArrayletObjectModel : public GC_ArrayletObjectModelBase
 						arrayletElementOffset = 0;
 					}
 					break;
+
+				default :
+					// array size other than 1/2/4/8 (mainly for flattened types)
+					{
+						U_8* srcCursor = ((U_8*)srcLeafAddress) + arrayletElementOffset * elementSize;
```

Comment:
Currently, these API's are only used for primitive array copies (i.e. jniGetPrimitiveArrayCritical) so the only allowable sizes are 1,2,4,8 bytes.

There is no spec on the JNI behaviour for valuetypes so right now dont know if we will even need a VT copy array helper for JNI. However, we might need one for Unsafe. 

In any case, VTs will need special handling for array copies since the elements have varying alignment constraints and may or may not contain references. For VT arrays we should create a separate helper that is optimized to recognize the cases where it can do a fast copy and when it must look at the instance description and do slot sized copies with the required barriers.

For now we should simply ensure that the primitive array copies behave correctly with the new stride calculation.

The default case in these APIs should assertUnimplemented (or assertUnreachable) since we currently do not expect anything other than primitive arrays to be using them.